src/tilesource.js

```diff
@@ -217,12 +226,49 @@ $.TileSource.prototype = /** @lends OpenSeadragon.TileSource.prototype */{
      * Subclasses should override this if tileSizes can be different at different levels
      *   such as in IIIFTileSource.  Code should use this function rather than reading
      *   from .tileSize directly.  tileSize may be deprecated in a future release.
+     * @deprecated
      * @function
      * @param {Number} level
      */
     getTileSize: function( level ) {
+        $.console.error(
+            "[TileSource.getTileSize] is deprecated." +
+            "Use TileSource.getTileWidth() and TileSource.getTileHeight() instead"
+        );
         return this.tileSize;
     },
+    
+    /**
+     * Return the tileWidth for a given level.
+     * Subclasses should override this if tileWidth can be different at different levels
+     *   such as in IIIFTileSource.  Code should use this function rather than reading
+     *   from .tileWidth directly.
+     * @function
+     * @param {Number} level
+     */
+    getTileWidth: function( level ) {
+        // If tileWidth was not set, fallback by setting it to tileSize
+        if( this.tileWidth === undefined){
```

Comment:
Supporting other TileSource subclasses was what I was trying to accomplish. But yeah, falling back to the deprecated getTileSize makes more sense.